test/RibbonThetaVault.js

```diff
@@ -2983,6 +2992,160 @@ function behavesLikeRibbonOptionsVault(params) {
+          );
+
+          // Share amount is burned
+          assert.equal(
+            (await this.vault.balanceOf(user)).toString(),
+            parseEther("0.9")
+          );
+
+          assert.equal(
+            (await this.vault.totalSupply()).toString(),
+            parseEther("0.9")
+          );
+
+          await expect(res)
+            .to.emit(this.vault, "Withdraw")
+            .withArgs(
+              user,
+              parseEther("0.1"),
+              parseEther("0.1"),
+              parseEther("0.0")
+            );
+        });
+      } else {
+        it("withdraw funds with no fee", async function () {
+          const depositAmount = BigNumber.from("100000000000");
+          await depositIntoVault(
+            params.collateralAsset,
+            this.vault,
+            depositAmount
+          );
+          await this.vault.connect(ownerSigner).sunset(this.v2vault.address);
+
+          const startAssetBalance = await this.assetContract.balanceOf(user);
+
+          const res = await this.vault.withdraw(BigNumber.from("10000000000"), {
+            gasPrice,
+          });
+          await res.wait();
+
+          assert.equal(
+            (await this.assetContract.balanceOf(this.vault.address)).toString(),
+            BigNumber.from("90000000000").toString()
+          );
+
+          assert.equal(
+            (await this.assetContract.balanceOf(user))
+              .sub(startAssetBalance)
+              .toString(),
+            BigNumber.from("10000000000").toString()
+          );
+
+          // Share amount is burned
+          assert.equal(
+            (await this.vault.balanceOf(user)).toString(),
+            BigNumber.from("90000000000")
+          );
+
+          assert.equal(
+            (await this.vault.totalSupply()).toString(),
+            BigNumber.from("90000000000")
+          );
+
+          await expect(res)
+            .to.emit(this.vault, "Withdraw")
+            .withArgs(
+              user,
+              BigNumber.from("10000000000"),
+              BigNumber.from("10000000000"),
+              BigNumber.from("0")
+            );
+        });
+      }
+
+      it("migrate calls V2 depositFor with correct address and amount", async function () {
+        // required for the next step to be able to fully withdraw
+        const minimumAmount = BigNumber.from(await this.vault.MINIMUM_SUPPLY());
+        await depositIntoVault(
+          params.collateralAsset,
+          this.vault,
+          minimumAmount
+        );
+
+        const depositAmount = BigNumber.from("10000000000");
+        await depositIntoVault(
+          params.collateralAsset,
+          this.vault,
+          depositAmount
+        );
+        await expect(await this.vault.balanceOf(user)).to.be.above(0);
+
+        await expect(this.vault.migrate()).to.be.revertedWith("Not sunset");
+
+        await this.vault.connect(ownerSigner).sunset(this.v2vault.address);
+
+        await this.vault.migrate();
```

Comment:
We need to test the [assertions for migrate](https://github.com/ribbon-finance/structured-products/pull/153/files#r675538371)